core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -608,11 +613,13 @@ func createChainWriter(
 	switch chainFamily {
 	case relay.NetworkSolana:
 		var solConfig chainwriter.ChainWriterConfig
-		if solana.PublicKeyLength != len(offrampProgramAddress) {
-			return nil, fmt.Errorf("invalid offrampProgramAddress length: %d", len(offrampProgramAddress))
+		var offrampProgramAddress solana.PublicKey
+		// NOTE: this function can still be called with EVM inputs, and PublicKeyFromBytes will panic on addresses with len=20
+		// technically we only need the writer to do fee estimation so this doesn't matter and we can use a zero address
```

Comment:
I guess this is a bit counter-intuitive, worth a potential re-organization of this code in future PRs.

Reply:
Yeah I agree, we were talking about extracting EstimateFee from chain writer. We could just always bind the zero address here since it doesn't really matter what the value is. On EVM we end up binding it to the address from the destination chain which doesn't exist on the source chain at all